pkg/cmd/cli/config_file.go

```diff
@@ -0,0 +1,99 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/viper"
+)
+
+// Note on use of config file:
+// - should implement a consistent way of reading all values
+// and harmonizing them with flags
+func (top *Options) determineUsageMode(rbacMode *bool) error {
+
+	if err := top.prepareViperConfig(); err != nil {
+		return err
+	}
+
+	// TODO(mitchdraft) - allow config to be overriden by flags
+	*rbacMode = viper.GetBool("enable_rbac_mode")
+	return nil
+}
+
+func (top *Options) determineVerbosity(verbose *bool) error {
+
+	if err := top.prepareViperConfig(); err != nil {
+		return err
+	}
+
+	// TODO(mitchdraft) - allow config to be overriden by flags
+	*verbose = viper.GetBool("verbose")
+	return nil
+}
+
+func writeDefaultConfigFile(fp string) error {
+	fmt.Printf("Squash config file not found. Writing default config to %v.\n", fp)
+	var defaultConfigYaml = []byte(`# Squash configuration file
+# The specification can be found at https://squash.solo.io
+enable_rbac_mode: false
+verbose: true
+createdby: squash-initialization
+`)
+	if err := ioutil.WriteFile(fp, defaultConfigYaml, 0644); err != nil {
+		return err
+	}
+	return nil
+}
+
+// This needs to be called before viper can read any config values
+func (top *Options) prepareViperConfig() error {
+	if top.Internal.ConfigLoaded {
+		// only load the config once
+		return nil
+	}
+	// read config file
+	// TODO(mitchdraft) - get this from an optional flag
+	cfgFile := ""
+	if cfgFile != "" {
+		// Use config file from the flag.
+		top.printVerbosef("Reading squash config from %v\n", cfgFile)
+		viper.SetConfigFile(cfgFile)
+	} else {
+		// Find home directory.
+		home, err := homedir.Dir()
+		if err != nil {
+			return err
+		}
+		squashDir := filepath.Join(home, ".squash")
+		if err := os.MkdirAll(squashDir, 0755); err != nil {
+			return err
+		}
+		squashConfigFile := filepath.Join(squashDir, "config.yaml")
+		if _, err := os.Stat(squashConfigFile); err == nil {
+			// path exists
+			top.printVerbosef("Reading squash config from %v\n", squashConfigFile)
+
+		} else if os.IsNotExist(err) {
```

Comment:
this if and the following else do the same thing.
why is the if needed?

Reply:
whoops, fixed